File: TUnit.TestProject.FSharp/ByteArgumentTests.fs
namespace TUnit.TestProject.FSharp

open TUnit.Core

// F# equivalent of ByteArgumentTests.cs

type ByteArgumentTests() =
    [<Test>]
    [<Arguments(1uy)>]
    member _.Normal(b: byte) =
        () // Dummy method

    [<Test>]
    [<Arguments(1uy)>]
    [<Arguments(null)>]
    member _.Nullable(b: byte option) =
        () // Dummy method

File: tests/TUnit.TestProject/ByteArgumentTests.cs
namespace TUnit.TestProject;

public class ByteArgumentTests
{
    [Test]
    [Arguments((byte) 1)]
    public void Normal(byte b)
    {
        // Dummy method
    }

    [Test]
    [Arguments((byte) 1)]
    [Arguments(null)]
    public void Nullable(byte? b)
    {
        // Dummy method
    }
}
